File: C++ Projects/Quiz/questiontf.cpp
//-------------------------
//class implentation(.cpp)
//author: Noah Funderburgh
//File:questiontf.cpp
//-------------------------
#include <cstring>
#include <iostream>
#include "question.h"
#include "questiontf.h"

using namespace std;
//------------------------
questionTF::~questionTF(){

}
//------------------------
questionTF::questionTF(){

}
//------------------------
void questionTF::showQuestion(){
    cout << questionText << " (True or False?)" <<  endl;
}
//------------------------
void questionTF::showAnswer(){
    cout << answerText << endl;
}
//------------------------
bool questionTF::checkAnswer(string answer){
    string answerTextLower = answerText;
    for(int i = 0;i<=answer.length();i++){
        if(answer[i] >= 65 && answer[i]<=92){
            answer[i]=answer[i]+32;
        }
    }

    for(int i = 0;i<=answerTextLower.length();i++){
        if(answerTextLower[i] >= 65 && answerTextLower[i]<=92){
            answerTextLower[i]=answerTextLower[i]+32;
        }
    }

    if(answer.at(0) == answerTextLower.at(0)){
        correct = true;
        return true;
    }
    else{
        correct = false;
        return false;
    }
}
//------------------------

File: Java Projects/Quiz/Questiontf.java

/**
* Implements classes need to add a True False question
* Functionality to our Quiz
*
* @author  Noah Funderburgh
* @version 1.0
* @since   3/3/2020
*/

public class Questiontf extends Question{
	/**
	 * This functions will show the answer when calling the function for true false questions.
   	 */
	void showQuestion() {
		System.out.println(questionText + " (True of False?)");
	}

//-------------------------------------------------------
	/**
	 * This function checks the if userAnswer is equal to Answer by making both strings lower case
	 * and then comparing the two to check if there equal.
	 * 
	 * @param userAnswer answer thats input by the user in the quiz
	 * @return boolean returns true if there equal else it returns false.
   	 */
	boolean checkAnswer(String userAnswer) {
		userAnswer = userAnswer.toLowerCase();
		answerText = answerText.toLowerCase();
		if(answerText.equals(userAnswer)) {
			return true;
		}
		if( answerText.contentEquals("true") && userAnswer.contentEquals("t") ) {
			return true;
		}
		if( answerText.contentEquals("false") && userAnswer.contentEquals("f") ) {
			return true;
		}
		else {
			return false;
		}
	}
}
